internal/services/dns/dns_cname_record_resource.go

```diff
@@ -191,6 +191,9 @@ func resourceDnsCNameRecordRead(d *pluginsdk.ResourceData, meta interface{}) err
 			targetResourceId := ""
 			if props.TargetResource != nil && props.TargetResource.Id != nil {
 				targetResourceId = *props.TargetResource.Id
+				if parsedID, err := recordsets.ParseRecordTypeIDInsensitively(targetResourceId); err == nil {
+					targetResourceId = parsedID.ID()
+				}
```

Comment:
@wuxu92 We still need to return the error here since we're expecting a record type ID. If the service suddenly begins returning an ID of a completely different type we would never know.
```suggestion
				parsedID, err := recordsets.ParseRecordTypeIDInsensitively(targetResourceId)
				if err != nil {
					return err
				}
				targetResourceId = parsedID.ID()
```

Reply:
Hi @wuxu92 - I don't think this is the correct way to handle the reported issue. The target ref is intended for only a handful of resource types:
<img width="247" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/11830746/3d16120a-accb-4d8a-ad80-b022f534445c">
I think the user should be using `record` not `target_resource_id` in the configuration? Can you try a repro that way to see if the issue is resolved that way? Since the `record` property is supposed to be another DNS record, this would be a more stable place to work with any casing mitigations if they are still required.